### DISCUSSION TITLE
How can I control the placement of glyphs during patch?

### DISCUSSION THREAD

**Discussion by rew1nter** [ryanoasis/nerd-fonts#1322] [General]
photo 1 shows the font I patched right now, and photo 2 is of a font that I patched last year. 

You can see that the powerline in photo 1 is placed upwards than the powerline in photo 2. I like the powerline placement in photo 2 more. The powerline of photo 2 is a bit bigger too. Is it possible to control the placement and size of these glyphs?
  
photo 1 - size 11:
![image](https://github.com/ryanoasis/nerd-fonts/assets/64508791/9c5e0b38-2e22-40ea-a83f-918c55a33d88)
photo 2 - size 11:
![image](https://github.com/ryanoasis/nerd-fonts/assets/64508791/4e35cb8e-b6ec-44f3-bc8b-97c3a688ac9e)

**Comment by Finii** [comment]
Hmm. Can you show pictures of both fonts again but also including a real Powerline glyph (e.g. codepoint `E0B0`) rather than the heavy bracket?

The heavy brackets have a rather complicated scale rule.

Are all other icons placed ok?

**Reply by Finii** [reply]
In principle the heavy brackets are controlled here:

![Screenshot_20230723_153721](https://github.com/ryanoasis/nerd-fonts/assets/16012374/ecbb456f-4a6a-4374-b510-cc804587a512)

Probably a bigger y-padding is what you want. Maybe 0.35. But that will not change the alignment.

**Reply by rew1nter** [reply]
ugh. How could it be aligned

**Reply by Finii** [reply]
It must be placed somewhere. For this is must be scaled and positioned. There is no such thing as a 'natural' scale or position.

**Comment by rew1nter** [comment]
photo 1, size - 11, current patch:
![image](https://github.com/ryanoasis/nerd-fonts/assets/64508791/255b3332-db8b-4382-8586-9e06c11a4fb8)

photo 2, size - 11, old patch:
![image](https://github.com/ryanoasis/nerd-fonts/assets/64508791/5d914707-2df9-412d-88f5-4d0c4e1d44af)

The glyph placement difference is more visible now. Although the size doesn't seem to differ with this glyph. 

Is it possible to manipulate the size as well during patch?

**Reply by Finii** [reply]
This is your old-patch-image:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/93e0afab-1b91-4d66-a9c9-0004dd857a75)

You see that the `E0B0` is smaller than the ascender of `f` and than the descender of `g`.
The Powerline glyphs should fill the complete 'cell', which is obviously not the case here.
It's also not the case with the new-patch.

There must be a different problem. Probably the letters and the Powerline glyph come from different font files. One would need to examine the concrete font file. If the original font is not broken it is impossible that this is the result of the patcher.

**Reply by Finii** [reply]
> Is it possible to manipulate the size as well during patch?

The size aims to fill the 'cell', i.e. original descender to ascender height. It does not really make sense to change that, except when the original font has wrong and/or confusing information. Some hints should be printed on patcher runs if you give `--debug 2`.

It is not forseen to have the Powerine glyphs anything but the line height (cell height).

Here an example how it is supposed to look like

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/60981d8f-b00e-4e6f-b4b9-91e546e7fdec)
_Taken from https://github.com/justjanne/powerline-go_

**Comment by rew1nter** [comment]
After investigation and going through old files, I can confirm you're correct. These are indeed two different fonts. 

I found the old font and it works as I want when patched. But the new one, which is the same font virtually, pushes these glyphs upwards after patch.

Thanks for helping me with this. I appreciate your time.
